File: core/src/main/java/br/ufs/nubankchallenge/core/infrastructure/models/ChargebackActionsPayload.java
package br.ufs.nubankchallenge.core.infrastructure.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import io.reactivex.annotations.Nullable;

/**
 * Created by @ubiratanfsoares
 */

public class ChargebackActionsPayload {

    @Nullable public String title;
    @Nullable @SerializedName("comment_hint") public String hint;
    @Nullable @SerializedName("reason_details") public List<ReasonDetailPayload> reasons;
    public boolean autoblock;

    public static class ReasonDetailPayload {
        public String id;
        public String title;
    }

}

File: core/src/main/java/br/ufs/nubankchallenge/core/infrastructure/models/ChargebackActionsPayload.kt
package br.ufs.nubankchallenge.core.infrastructure.models

import com.google.gson.annotations.SerializedName

/**
 * Created by @ubiratanfsoares
 */

class ChargebackActionsPayload {

    var title: String? = null
    @SerializedName("comment_hint") var hint: String? = null
    @SerializedName("reason_details") var reasons: List<ReasonDetailPayload>? = null
    var autoblock: Boolean = false

    class ReasonDetailPayload {
        var id: String? = null
        var title: String? = null
    }

}


